src/sbvr-api/sbvr-utils.coffee

```diff
@@ -687,34 +765,38 @@ exports.handleODataRequest = handleODataRequest = (req, res, next) ->
 		# Parse the OData requests
 		mapSeries body, (bodypart) ->
 			uriParser.parseOData(bodypart)
-			.then controlFlow.liftP (request) ->
-				# Get the full hooks list now that we can. We clear the hooks on
-				# the global req to avoid duplication
-				req.hooks = {}
-				request.hooks = getHooks(request)
-				# Add/check the relevant permissions
-				runHook('POSTPARSE', { req, request, tx: req.tx })
-				.return(request)
-				.then (uriParser.translateUri)
-				.then (request) ->
-					# We defer compilation of abstract sql queries with references to other requests
-					if request.abstractSqlQuery? && !request._defer
-						try
-							request.sqlQuery = memoizedCompileRule(request.abstractSqlQuery)
-						catch err
-							api[apiRoot].logger.error('Failed to compile abstract sql: ', request.abstractSqlQuery, err, err.stack)
-							throw new SqlCompilationError(err)
-					return request
-
-			.then (request) ->
-				# Run the request in its own transaction
-				runTransaction req, (tx) ->
-					if _.isArray request
-						env = new Map()
-						Promise.reduce(request, runChangeSet(req, res, tx), env)
-						.then (env) -> Array.from(env.values())
-					else
-						runRequest(req, res, tx, request)
+			.then (odata) ->
+				requestEnv = prepareEnv(req, odata)
+				Promise.resolve(requestEnv)
```

Comment:
It would be much nicer to do this as:
```
.then (odata) ->
	prepareEnv(req, odata)
.then controlFlow.liftP (requestEnv) ->

Reply:
I agree, however I need to declare it to make sure that `requestEnv` is in scope for the final `tapCatch` block (which also forces me to indent the next two `then` blocks). 
I don't like this that much (I will at least add a comment there, as the meaning is not clear and it might be changed by mistake in the future) however I would like to change it if you have any idea on how to make it better

Reply:
What about
```coffee
.then (odata) ->
	prepareEnv(req, odata)
.tap controlFlow.liftP (requestEnv) ->
```
then?

Reply:
This still doesn't work as expected. If there is an error in the `tap` block, then `requestEnv` will be undefined in the in `tapCatch` that comes after. We need to declare it as a variable that exists outside the promise chain that does the compilation and execution in order to be able to refer to it in the `tapCatch` block.

Reply:
What about
```coffee
.then (odata) ->
	requestEnv = prepareEnv(req, odata)
	controlFlow.liftP (requestEnv) ->
```
then? It'll still avoid the `Promise.resolve` at least